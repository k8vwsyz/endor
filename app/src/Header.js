import { AppBar, Button, IconButton, Menu, MenuItem, Toolbar, Typography } from '@material-ui/core';
import { makeStyles, useTheme } from "@material-ui/core/styles";
import useMediaQuery from "@material-ui/core/useMediaQuery";
import MenuIcon from '@material-ui/icons/Menu';
import React from 'react';
import { withRouter } from "react-router-dom";

const useStyles = makeStyles((theme) => ({
  button: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  }
}));

const Header = props => {
  const { history } = props;
  const classes = useStyles();
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("xs"));
  const [anchorEl, setAnchorEl] = React.useState(null);
  const open = Boolean(anchorEl);

  const menuItems = [
    {
      key: 1,
      title: "Home",
      pageURL: "/"
    },
    {
      key: 2,
      title: "About",
      pageURL: "/about"
    },
    {
      key: 3,
      title: "News",
      pageURL: "/news"
    }
  ]

  const handleMenu = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleMenuClick = pageURL => {
    history.push(pageURL);
    setAnchorEl(null);
  };

  const handleButtonClick = pageURL => {
    history.push(pageURL);
  };

  const NavLinks = () => (
    <>
      {menuItems.map(menuItem => {
        const { key, title, pageURL } = menuItem;
        return (
          <Button
            key={key.toString()}
            variant="outlined"
            onClick={() => handleButtonClick(pageURL)}
            className={classes.button}
            color="inherit">
            {title}
          </Button>
        );
      })}
    </>
  )

  const MobileNav = () => (
    <>
      <IconButton
        edge="start"
        color="inherit"
        aria-label="menu"
        onClick={handleMenu}
      >
        <MenuIcon />
      </IconButton>
      <Menu
        id="menu-appbar"
        anchorEl={anchorEl}
        anchorOrigin={{
          vertical: "top",
          horizontal: "right"
        }}
        keepMounted
        transformOrigin={{
          vertical: "top",
          horizontal: "right"
        }}
        open={open}
        onClose={() => setAnchorEl(null)}>
        {menuItems.map(menuItem => {
          const { key, title, pageURL } = menuItem;
          return (
            <MenuItem
              key={key.toString()}
              onClick={() => handleMenuClick(pageURL)}>
              {title}
            </MenuItem>
          );
        })}
      </Menu>
    </>
  )

  return (
    <AppBar position="static">
      <Toolbar>
        <Typography variant="h6" className={classes.title}>
          Header
        </Typography>
        {isMobile ? (MobileNav()) : (NavLinks())}
      </Toolbar>
    </AppBar>
  );
}

export default withRouter(Header);