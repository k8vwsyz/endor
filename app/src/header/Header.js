import React from 'react';
import { AppBar, Button, IconButton, Menu, MenuItem, Toolbar, Typography } from '@material-ui/core';
import { makeStyles, useTheme } from "@material-ui/core/styles";
import useMediaQuery from "@material-ui/core/useMediaQuery";
import MenuIcon from '@material-ui/icons/Menu';
import { useHistory } from "react-router-dom";

const useStyles = makeStyles((theme) => ({
  button: {
    marginLeft: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
    color: 'Black',
  },
  whiteAppbar:{
    backgroundColor: 'White',
    color:'Black',
  }
}));

const NavLinks = props => {
  const { menuItems, handleButtonClick, classes } = props;

  return (
    <>
      {menuItems.map(menuItem => {
        const { key, title, pageURL } = menuItem;
        return (
          <Button
            key={key.toString()}
            variant="outlined"
            onClick={() => handleButtonClick(pageURL)}
            className={classes.button}
            color="inherit">
            {title}
          </Button>
        );
      })}
    </>
  );
}

const MobileNav = props => {

  const { menuItems, handleMenuClickCallback } = props;

  const [anchorEl, setAnchorEl] = React.useState(null);
  const open = Boolean(anchorEl);

  const handleMenu = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleMenuClick = pageURL => {
    handleMenuClickCallback(pageURL);
    setAnchorEl(null);
  }

  return (
    <>
      <IconButton
        edge="start"
        color="inherit"
        aria-label="menu"
        onClick={handleMenu}
      >
        <MenuIcon />
      </IconButton>
      <Menu
        id="menu-appbar"
        anchorEl={anchorEl}
        anchorOrigin={{
          vertical: "top",
          horizontal: "right"
        }}
        keepMounted
        transformOrigin={{
          vertical: "top",
          horizontal: "right"
        }}
        open={open}
        onClose={() => setAnchorEl(null)}>
        {menuItems.map(menuItem => {
          const { key, title, pageURL } = menuItem;
          return (
            <MenuItem
              key={key.toString()}
              onClick={() => handleMenuClick(pageURL)}>
              {title}
            </MenuItem>
          );
        })}
      </Menu>
    </>
  )
}

const Header = props => {
  const { menuItems } = props;
  const classes = useStyles();
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("xs"));
  const history = useHistory();

  const handleMenuClick = pageURL => {
    history.push(pageURL);
  };

  const handleButtonClick = pageURL => {
    history.push(pageURL);
  };

  return (
    <AppBar className={classes.whiteAppbar} position="static" >
      <Toolbar>
        <Typography variant="h6" className={classes.title}>
          Agile Endor
        </Typography>
        {isMobile ? (
          <MobileNav
            menuItems={menuItems}
            handleMenuClickCallback={handleMenuClick} />
        ) : (
            <NavLinks
              menuItems={menuItems}
              handleButtonClick={handleButtonClick}
              classes={classes} />
          )}
      </Toolbar>
    </AppBar>
  );
}

export default Header;
