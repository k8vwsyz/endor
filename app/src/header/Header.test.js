import React from "react";
import { render, screen, cleanup } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Header from "./Header";
import { MemoryRouter, Route } from "react-router-dom";

afterEach(cleanup);

const menuItems = [
  {
    key: 1,
    title: "Home",
    pageURL: "/"
  },
  {
    key: 2,
    title: "About",
    pageURL: "/about"
  }
];


it('navigate to page', () => {
  let history_, location_

  render(
    <MemoryRouter initialEntries={["/"]}>
      <Header menuItems={menuItems} />
      <Route
        path="*"
        render={({ history, location }) => {
          history_ = history;
          location_ = location;
          return null;
        }}
      />
    </MemoryRouter>
  )

  expect(location_.pathname).toBe("/")

  const aboutButtton = screen.getByText("About")

  userEvent.click(aboutButtton)

  expect(location_.pathname).toBe("/about")
});

it('snapshot test of page', () => {

  const { asFragment } = render(
    <MemoryRouter initialEntries={["/"]}>
      <Header menuItems={menuItems} />
    </MemoryRouter>
  )

  expect(asFragment()).toMatchSnapshot()
});
