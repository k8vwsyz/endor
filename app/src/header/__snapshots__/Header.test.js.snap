// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test of page 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-whiteAppbar-6 MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        class="MuiTypography-root makeStyles-title-5 MuiTypography-h6"
      >
        Agile Endor
      </h6>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-4 MuiButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Home
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-4 MuiButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          About
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</DocumentFragment>
`;
